internal/services/apicenter/api_center_environment_resource.go

```diff
@@ -0,0 +1,304 @@
+				Name:       &model.Identification,
+				Properties: &apiCenterEnvironmentProps,
+				Type:       &model.Type,
+			}
+
+			if _, err = client.CreateOrUpdate(ctx, id, apiCenterEnvironment); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r ApiCenterEnvironmentResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ApiCenter.EnvironmentsClient
+			id, err := environments.ParseEnvironmentID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var state ApiCenterEnvironmentResourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil || existing.Model == nil {
+				return fmt.Errorf("reading %s: %v", id, err)
+			}
+
+			model := existing.Model
+
+			if metadata.ResourceData.HasChange("description") {
+				model.Properties.Description = pointer.To(state.Description)
+			}
+
+			if metadata.ResourceData.HasChange("development_portal_uri") {
+				model.Properties.Onboarding.DeveloperPortalUri = pointer.To([]string{state.DevPortalUri})
+			}
+
+			if metadata.ResourceData.HasChange("server_type") {
+				model.Properties.Server.Type = pointer.To(environments.EnvironmentServerType(state.ServerType))
+			}
+
+			if metadata.ResourceData.HasChange("management_portal_uri") {
+				model.Properties.Server.ManagementPortalUri = pointer.To([]string{state.MgmtPortalUri})
+			}
+
+			if metadata.ResourceData.HasChange("instructions") {
+				model.Properties.Onboarding.Instructions = pointer.To(state.Instructions)
+			}
+
+			if _, err = client.CreateOrUpdate(ctx, *id, *existing.Model); err != nil {
+				return fmt.Errorf("updating %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r ApiCenterEnvironmentResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ApiCenter.EnvironmentsClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			id, err := environments.ParseEnvironmentID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(existing.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := ApiCenterEnvironmentResourceModel{
+				ServiceId: services.NewServiceID(subscriptionId, id.ResourceGroupName, id.ServiceName).ID(),
+			}
+			if model := existing.Model; model != nil {
+				state.Identification = pointer.From(model.Name)
+				if props := existing.Model.Properties; props != nil {
+					state.Name = props.Title
```

Comment:
Ditto here, `identification` -> `name` and `name` -> `title`
```suggestion
				state.Identification = pointer.From(model.Name)
				if props := existing.Model.Properties; props != nil {
					state.Name = props.Title
```